### DISCUSSION TITLE
How to Mark All Selected Books as Read

### DISCUSSION THREAD

**Discussion by pjkaufman** [koreader/koreader#12915] [Q&A] [answered]
I use KOReader to help keep track of which books I have read via the file system (I could use Calibre, but that forces a particular structure I do not adhere to). I use KOReader on Android and PC, so I would prefer to stick with that solution for managing my library. I know it is possible to select multiple books at one time, but I noticed that there is no option to mark as finished when selecting multiple books. I would like to have the ability to do this. I know that this likely can be added by a custom snippet or plugin.

Is this feature something I just missed in the options somewhere when I selected multiple files? If not, is there a particular place I should start looking at to add that option to KOReader via a plugin or otherwise?

Thanks for helping me understand how to proceed.

**Comment by shuvashish76** [comment, marked as answer]
Cross reference
- https://www.mobileread.com/forums/showthread.php?t=366967
- https://www.mobileread.com/forums/showthread.php?t=368850

I've mentioned my way of doing it in my #13442 comment.

**Reply by pjkaufman** [reply]
Thanks for the heads up. It looks like it was added as a feature. I was thinking of adding it myself if need be. But I don't know enough Lua to be certain how to do it well.

I am glad it was added.
